src/main/java/org/mariadb/jdbc/internal/com/read/OkPacket.java

```diff
@@ -0,0 +1,150 @@
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this list
+ * of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice, this
+ * list of conditions and the following disclaimer in the documentation and/or
+ * other materials provided with the distribution.
+ *
+ * Neither the name of the driver nor the names of its contributors may not be
+ * used to endorse or promote products derived from this software without specific
+ * prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS  AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+ * OF SUCH DAMAGE.
+ *
+ */
+
+package org.mariadb.jdbc.internal.com.read;
+
+import java.nio.charset.StandardCharsets;
+
+import org.mariadb.jdbc.internal.MariaDbServerCapabilities;
+
+public class OkPacket {
+
+  private final long affectedRows;
+  private final long insertId;
+  private final short statusFlags;
+  private final short warnings;
+  private final String info;
+  private final String sessionStateInfo;
+
+  /**
+   * Read Ok stream result.
+   *
+   * @param buffer current stream's rawBytes
+   */
+  public OkPacket(Buffer buffer, long clientCapabilities) {
+
+    buffer.skipByte(); // fieldCount
+    affectedRows = buffer.getLengthEncodedNumeric();
+    insertId = buffer.getLengthEncodedNumeric();
+ 
+    if ((clientCapabilities & MariaDbServerCapabilities.CLIENT_PROTOCOL_41) !=0) {
+    	statusFlags = buffer.readShort();
+    	warnings = buffer.readShort();
+    }
+    else if ((clientCapabilities & MariaDbServerCapabilities.TRANSACTIONS) !=0) {
+    	statusFlags = buffer.readShort();
+    	warnings = 0;
+    } 
+    else {
+    	statusFlags = 0;
+    	warnings = 0;
+    }
+    
+    String message = "";
+    String sessionStateMessage = "";
+    if (buffer.remaining() > 0) {
+    	if ((clientCapabilities & MariaDbServerCapabilities.CLIENT_SESSION_TRACK) !=0) {
+    		message = buffer.readStringLengthEncoded(StandardCharsets.UTF_8);
+    		
+        	if ((statusFlags & MariaDbServerCapabilities.SERVER_SESSION_STATE_CHANGED) !=0)  {
```

Comment:
```suggestion
        	if ((statusFlags & ServerStatus.SERVER_SESSION_STATE_CHANGED) !=0)  {
```

(as per comment about this constant above)